packages/concerto-metamodel/lib/metamodel.cto

```diff
@@ -135,6 +135,20 @@ concept StringRegexValidator {
   o String flags
 }
 
+concept MapDeclaration extends Declaration {
+  o Boolean isAbstract default=false
+  o Identified identified optional
+  o TypeIdentifier superType optional
+	o MapProperty key 
+	o MapProperty value 
+}
+
+concept MapProperty {
```

Comment:
Let's not call this a "property" if it doesn't extend Property. This is a different animal...

Reply:
This is now refactored so that `MapKey` and `MapValue` inherit from `Declaration`

```
concept MapDeclaration extends Declaration {
  o MapKey key
  o MapValue value
}

concept MapKey extends Declaration {
  o Property property
  o TypeIdentifier type
}

concept MapValue extends Declaration {
  o Property property
  o TypeIdentifier type
 }
```

Which might be further refactored to  down to a type of

```
concept MapProperty extends Declaration  {
  o Property property
  o TypeIdentifier type
}

concept MapKey extends MapProperty { }

concept MapValue extends MapProperty { }
```

or perhaps just

```
concept MapProperty extends Declaration  {
  o Property property
  o TypeIdentifier type
}

concept MapDeclaration extends Declaration {
  o MapProperty key
  o MapProperty value
}
```


Conceptually do you feel like this is headed in the right direction, or am I headed into the weeds here? Based on your above comment unsure if you have something else in mind entirely eg: "Lets not call this a property" - but to my mind a Key and Value are both properties of a Map Declaration. Then again, `MapProperty extends Declaration` seems incorrect (Property extending Declaration?) - this could be better named `MapPropertyDeclaration`, though this sounds a little overloaded.